src/Microsoft.AspNet.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -236,6 +279,78 @@ public ForChunkedEncoding(bool keepAlive, FrameContext context)
                 return 0;
             }
 
+            public override async Task<int> SkipAsyncImplementation(CancellationToken cancellationToken)
```

Comment:
Feels like all of the duplication can be removed by just passing a delegate (though the out parameter won't work :smile: )